gdpr/impl.go

```diff
@@ -69,12 +82,48 @@ func (p *permissionsImpl) allowSync(ctx context.Context, vendorID uint16, consen
 	return false, nil
 }
 
-type alwaysAllow struct{}
+func (p *permissionsImpl) allowPI(ctx context.Context, vendorID uint16, consent string) (bool, error) {
+	// If we're not given a consent string, respect the preferences in the app config.
+	if consent == "" {
+		return p.cfg.UsersyncIfAmbiguous, nil
+	}
+
+	parsedConsent, err := vendorconsent.ParseString(consent)
```

Comment:
Outside the scope of this PR perhaps, but do we always re-parse the entire string every time we check for a specific permission in it? I vaguely remember us having a "parsed GDPR" struct, but I may be mistaken.

Reply:
That is a good point. We would reparse the consent string for every bidder. It would change the method signature from the other allow methods, but would save a few cycles.